lib/commands/connect/shared.js

```diff
@@ -109,77 +110,69 @@ let request = exports.request = function(token, method, path, data){
   return promise;
 };
 
-let withUserConnections = exports.withUserConnections = function(token, appName, flags, allowNone, heroku){
+let withUserConnections = exports.withUserConnections = co.wrap(function* (token, appName, flags, allowNone, heroku){
   global_config();
 
   let resource_id = null;
   if (flags && flags.resource) {
     console.log("Calling Platform API for resouce id");
-    return Q.Promise(function(resolve, reject) {
-      return heroku.apps(appName).addons(flags.resource).info(function(err, addon) {
-        if(err) {
-          if (err.statusCode === 404) {
-            console.log("Connection with resource name '" + flags.resource + "' not found");
-            reject();
-          } else {
-            console.log(err);
-            reject(err);
-          }
+
+    return co(function* () {
+      try {
+        let addon = yield heroku.apps(appName).addons(flags.resource).info();
+      } catch(err) {
+        if (err.statusCode === 404) {
+          return yield Promise.reject("Connection with resource name '" + flags.resource + "' not found");
```

Comment:
I also don't know the correct way to do rejections. The docs seem pretty light on rejections, unfortunately. It seems to work identically, whether I use `return yield ...`, `throw yield ...` or just `yield ...`. I used `return` here just to make sure the function doesn't continue lumbering on, because it seemed safer to me, but if it's not necessary, I'm happy to remove it. `throw` feels semantically correct to me, but I couldn't find a single usage of `throw yield Promise.reject()`, so I have a feeling it's not the right thing to do.